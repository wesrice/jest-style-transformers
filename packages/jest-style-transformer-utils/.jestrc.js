const tsPreset = require('ts-jest/presets/js-with-babel/jest-preset');

module.exports = {
  ...tsPreset,
};
