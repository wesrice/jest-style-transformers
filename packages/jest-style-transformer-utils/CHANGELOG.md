# Changelog

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [jest-style-transformer-utils-v2.0.1](https://github.com/wesrice/jest-style-transformers/compare/jest-style-transformer-utils-v2.0.0...jest-style-transformer-utils-v2.0.1) (2021-09-17)


### Bug Fixes

* update dependencies that depend on `jest-style-transformer-utils` ([e50de9f](https://github.com/wesrice/jest-style-transformers/commit/e50de9f3094197d5fb8047452461935bfd6b2fb5))

# [jest-style-transformer-utils-v2.0.0](https://github.com/wesrice/jest-style-transformers/compare/jest-style-transformer-utils-v1.0.0...jest-style-transformer-utils-v2.0.0) (2021-09-15)


### Bug Fixes

* update cache busting parameter for release ([4800d2d](https://github.com/wesrice/jest-style-transformers/commit/4800d2d2841138f8c48c9190de9009b03641d21a))
* update dependencies ([909d499](https://github.com/wesrice/jest-style-transformers/commit/909d4990f37726c61509c1dbb8070cfd5def7dfb))
* updated cache key generation for Jest 27 compatibility ([5209876](https://github.com/wesrice/jest-style-transformers/commit/520987643f16c285892992f75e119de238b184e6))


### BREAKING CHANGES

* Utilizes api changed in Jest 27.

# jest-style-transformer-utils-v1.0.0 (2020-11-11)


### Features

* initial release ([776b06b](https://github.com/wesrice/jest-style-transformers/commit/776b06bd58e1e61268e0d91e77e312a9d26c1f9a))
