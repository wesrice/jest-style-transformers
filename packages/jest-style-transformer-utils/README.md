# Jest Style Transformer - Utils

This package isn't meant to have much utility by itself, but provides common
functionality to other `jest-style-transformer-*` packages.
