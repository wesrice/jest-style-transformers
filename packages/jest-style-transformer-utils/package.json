{
  "author": "Wes Rice",
  "dependencies": {},
  "description": "Utilities used by `jest-style-transformer-*` packages",
  "devDependencies": {
    "@types/cross-spawn": "^6.0.2",
    "@types/postcss-load-config": "^2.0.1",
    "jest": "^27.0.6",
    "postcss": "^8.3.5",
    "postcss-load-config": "^3.1.0",
    "ts-jest": "^27.0.3",
    "typescript": "^4.3.4"
  },
  "engines": {
    "node": ">= 12.0.0"
  },
  "files": [
    "build"
  ],
  "keywords": [
    "jest",
    "testing",
    "styles",
    "css",
    "css modules",
    "css-in-js"
  ],
  "license": "MIT",
  "main": "build/index.js",
  "name": "jest-style-transformer-utils",
  "peerDependencies": {
    "jest": "^27.0.6"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/wesrice/jest-style-transformers.git",
    "directory": "packages/jest-style-transformer-utils"
  },
  "scripts": {
    "build": "rm -rf build && tsc -p tsconfig.build.json",
    "registry:unpublish": "npm unpublish -f --registry http://localhost:4873/ jest-style-transformer-utils",
    "test": "jest --config .jestrc.js --passWithNoTests"
  },
  "types": "build/index.d.ts",
  "version": "2.0.1"
}
