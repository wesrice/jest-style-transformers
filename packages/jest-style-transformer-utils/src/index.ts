import { processor as postcssProcessor } from './postcss/processor';
import { runner as postcssRunner } from './postcss/runner';

export * from './createCacheKey';
export * as recorders from './recorders';
export { postcssProcessor, postcssRunner };
