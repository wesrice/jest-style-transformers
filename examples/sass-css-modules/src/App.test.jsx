import './App';
import { recorders } from 'jest-style-transformer-utils';

describe('jest-style-transformer-sass example', () => {
  it('style extraction does not regress', async () => {
    expect.assertions(1);
    expect(recorders.styles.get()).toMatchSnapshot();
  });
});
