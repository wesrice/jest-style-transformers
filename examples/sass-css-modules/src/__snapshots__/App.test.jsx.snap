// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-style-transformer-sass example style extraction does not regress 1`] = `
".src-a-1rydrMvuBG {
  color: white;
}.src-h1-3-5U7Q0Yy8 {
  background: black;
}"
`;
